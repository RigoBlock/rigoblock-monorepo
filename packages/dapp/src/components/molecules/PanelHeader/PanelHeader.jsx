import './PanelHeader.scss'
import PropTypes from 'prop-types'
import React from 'react'
import Tooltip from '../../atoms/Icon'

const PanelHeader = ({ tooltip, title }) => {
  const tooltipComponent = tooltip && (
    <Tooltip type={'help'} tooltipText={tooltip} />
  )

  return (
    <div className="panel-header">
      <h1>{title}</h1>
      {tooltipComponent}
    </div>
  )
}

PanelHeader.propTypes = {
  title: PropTypes.string.isRequired,
  tooltip: PropTypes.string
}

PanelHeader.defaultProps = {
  tooltip: null
}

export default PanelHeader
