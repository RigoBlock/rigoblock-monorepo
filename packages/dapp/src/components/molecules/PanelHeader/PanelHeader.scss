@import "../../_settings/variables.scss";

.panel-header {
  display: flex;
  padding: $dist-s;
  h1 {
    flex-grow: 1;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.17;
    letter-spacing: normal;
  }
}
