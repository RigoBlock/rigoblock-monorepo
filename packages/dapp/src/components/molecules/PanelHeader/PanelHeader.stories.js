import React from 'react'
import { storiesOf } from '@storybook/react'
import PanelHeader from './PanelHeader'
import '../../_settings/_base.scss'

storiesOf('Molecules/PanelHeader', module)
  .addDecorator(story => (
    <div
      style={{
        width: '400px',
        background: 'white',
        border: '1px solid rgba(0, 0, 0, 0.1'
      }}
    >
      {story()}
    </div>
  ))
  .add('default', () => <PanelHeader title="PanelHeader" />)
  .add('tooltip', () => (
    <PanelHeader title="PanelHeader" tooltip="empty tooltip" />
  ))
