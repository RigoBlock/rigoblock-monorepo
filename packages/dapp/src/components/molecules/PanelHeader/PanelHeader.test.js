import { shallow } from 'enzyme'
import PanelHeader from './PanelHeader.jsx'
import toJson from 'enzyme-to-json'

const props = { title: 'Test Header', tooltip: 'empty tooltip' }

describe('PanelHeader component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(PanelHeader, props)))
    ).toMatchSnapshot()
  })
})
