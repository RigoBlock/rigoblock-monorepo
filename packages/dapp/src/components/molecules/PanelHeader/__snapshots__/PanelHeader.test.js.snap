// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelHeader component renders correctly 1`] = `
<div
  className="panel-header"
>
  <h1>
    Test Header
  </h1>
  <Icon
    className=""
    onClick={[Function]}
    size="medium"
    tooltipText="empty tooltip"
    type="help"
  />
</div>
`;
